lib/kafka-consumer.js

```diff
@@ -303,6 +303,16 @@ KafkaConsumer.prototype.incrementalAssign = function(assignments) {
   return this;
 };
 
+/**
+ * Returns true if currently consumer is rebalancing, false otherwise.
+ *
+ * @return {Boolean}
+ */
+
+ KafkaConsumer.prototype.isRebalancing = function() {
+  return this._client.isRebalancing();
```

Comment:
I need to fix the Windows build error:

```bash
c:\sixfold\code\messaging-utils\node_modules\@sixfold\node-rdkafka\src\kafka-consumer.cc(390): error C2039: 'is_rebalancing': is not a member of
 'RdKafka::KafkaConsumer' [C:\sixfold\code\messaging-utils\node_modules\@sixfold\node-rdkafka\build\node-librdkafka.vcxproj]
  c:\sixfold\code\messaging-utils\node_modules\@sixfold\node-rdkafka\deps\include\rdkafkacpp.h(2479): note: see declaration of 'RdKafka::KafkaCo
  nsumer'
```

The hackish way would be adding `virtual bool is_rebalancing () = 0;` into `rdkafkacpp.h` file which comes with the prebuilt sources that is fetched from nuget registry.

But I want to understand where is this `isRebalancing` implementation. 

`this._client` refers to `this._client = new SubClientType(globalConf, topicConf);`
What is this `SubClientType`?

@poolik @emelyanovtv

Reply:
@lkallas `isRebalancing` is a method we added, that's why we have this "windows not building" issue in the first place:
1) We forked librdkafka to expose something that wasn't exposed through public APIs at the moment
2) node-rdkafka uses git submodule to build librdkafka and its own native bindings
3) we updated submodule to point to our for, BUT as on Windows it's not building librdkarfka then it won't compile

See https://github.com/6fold/librdkafka/pull/1/files (and our attempt to get this upstream: https://github.com/edenhill/librdkafka/pull/3593)

Reply:
implementation is [over here](https://github.com/6fold/librdkafka/pull/1/files#diff-84ac6df9f4f6c13f1f4eecf3b9c0e2cee4d7c4c005feea512ff3f82c000fb84eR2797). 
`
Regarding this._client and `SubClientType` it can be either [Kafka.Producer](https://github.com/6fold/node-rdkafka#sending-messages) or [Kafka.KafkaConsumer](https://github.com/6fold/node-rdkafka#kafkakafkaconsumer). 

I hope I answered to your questions.

Reply:
Ah, now I see. Was looking at the wrong `librdkafka` branch the whole time.

Have Visual Studio 2015 installed and vcpkg for installing correct dependencies.
Currently on my machine builds fail. But found out edenhill uses CI tools to build artifacts and then build nuget package from those artifacts: https://github.com/edenhill/librdkafka/tree/master/packaging/nuget